pioinstaller/core.py

```
@@ -116,28 +114,69 @@ def _install_platformio_core(shutdown_piohome=True, develop=False, ignore_python
-    except Exception as e:  # pylint:disable=broad-except
+    except subprocess.CalledProcessError as e:
         error = str(e)
         if util.IS_WINDOWS:
             error = (
                 "If you have antivirus/firewall/defender software in a system,"
                 " try to disable it for a while.\n %s" % error
             )
         raise exception.PIOInstallerException(
-            "Could not install pioarduino Core: %s" % error
+            "Could not install pioarduino Core with uv: %s" % error
         )
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)